form/data_transformers.rst

```diff
@@ -80,6 +80,7 @@ class::
                 ->addModelTransformer(new CallbackTransformer(
                     function ($tagsAsArray) {
```

Comment:
Another idea, if it's expected to be null or an array, we could add a typehint here `?array $tagsAsArray` and below `return implode(', ', $tagsAsArray ?? []);`, what do you think?

same for reversed transformation `?string $tagsAsString`...

Reply:
also the reversed transformation should take into account a possible null value to not end up with `[0 => ""]` in your DB.

Reply:
There should not be type hints in callbacks, but the data should be tested indeed:
```php
$builder
    ->addModelTransformer(new CallbackTransformer(
        function ($tagsAsArray) {
            if (null === $tagsAsArray) {
                return '';
            }

            // here we could also throw a `TransformationFailedException` if not an array

            return implode(...);
        },
        function ($tagsAsString) {
            if (null === $tagsAsString || '' === $tagsAsString) {
                return [];
            }

            // here we could also throw a `TransformationFailedException`  if not a string

            return explode(...);
        })
;
```

Reply:
So, we can put this :

` // Tags
        $builder->add('tags', TextType::class);
            $builder->get('tags')
                ->addModelTransformer(new CallbackTransformer(
                    function ($tagsAsArray) {
                        if (null === $tagsAsArray) {
                            return '';
                        }
                        // transform the array to a string
                        // transform the original value into a format that'll be used to render the field
                        return implode(', ', $tagsAsArray);
                    },
                    function ($tagsAsString) {
                        if (null === $tagsAsString || '' === $tagsAsString) {
                            return [];
                        }
                        // transform the string back to an array
                        // it transforms the submitted value back into the format you will use
                        return explode(', ', $tagsAsString);
                    }
                ))

        ;`

Reply:
@yceruto @HeahDude

Reply:
I would also throw a `TransformationFailedException` as suggested in https://github.com/symfony/symfony-docs/pull/12937#discussion_r368075339.